packages/amplify_api/android/src/test/kotlin/com/amazonaws/amplify/amplify_api/GraphQLApiUnitTests.kt

```diff
@@ -0,0 +1,299 @@
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazonaws.amplify.amplify_api
+
+import com.amazonaws.amplify.amplify_api.types.FlutterApiErrorMessage
+import com.amplifyframework.api.ApiCategory
+import com.amplifyframework.api.ApiException
+import com.amplifyframework.api.aws.GsonVariablesSerializer
+import com.amplifyframework.api.graphql.GraphQLOperation
+import com.amplifyframework.api.graphql.GraphQLRequest
+import com.amplifyframework.api.graphql.GraphQLResponse
+import com.amplifyframework.api.graphql.SimpleGraphQLRequest
+import com.amplifyframework.core.Amplify
+import com.amplifyframework.core.Consumer
+import io.flutter.plugin.common.MethodCall
+import io.flutter.plugin.common.MethodChannel
+import org.junit.Assert
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mockito.*
+import org.robolectric.RobolectricTestRunner
+import java.lang.reflect.Field
+import java.lang.reflect.Modifier
+
+@RunWith(RobolectricTestRunner::class)
+class GraphQLApiUnitTests {
+    lateinit var flutterPlugin: AmplifyApiPlugin
+    private var mockApi = mock(ApiCategory::class.java)
+    private val mockResult: MethodChannel.Result = mock(MethodChannel.Result::class.java)
+    private val mockGraphQLOperation = mock(GraphQLOperation::class.java)
+
+    @Before
+    fun setup() {
+        flutterPlugin = AmplifyApiPlugin()
+        setFinalStatic(Amplify::class.java.getDeclaredField("API"), mockApi)
+    }
+
+    @Test
+    fun test_query_returns_success() {
+        val testRequest = HashMap<String, Any>();
+        testRequest["document"] = """query MyQuery {
+            listBlogs {
+                items {
+                id
+                name
+                createdAt
+                }
+            }
+        }"""
+        testRequest["variables"] = HashMap<String, Any>();
+
+        val graphQLResponse: GraphQLResponse<String> = GraphQLResponse("result", null)
+
+        doAnswer { invocation ->
+            Assert.assertEquals(
+                    SimpleGraphQLRequest<String>(
+                            testRequest["document"] as String,
+                            testRequest["variables"] as Map<String, Any>,
+                            String::class.java,
+                            GsonVariablesSerializer()
+                    ),
+                    invocation.arguments[0]
+            )
+            (invocation.arguments[1] as Consumer<GraphQLResponse<String>>).accept(
+                    graphQLResponse
+            )
+            mockGraphQLOperation
+        }.`when`(mockApi).query(
+                any<GraphQLRequest<String>>(),
+                any(),
+                any())
+
+        flutterPlugin.onMethodCall(
+                MethodCall("query", testRequest),
+                mockResult
+        )
+
+        verify(mockResult, times(1)).success(mapOf(
```

Comment:
Drop `times(1)` throughout the PR. `verify(foo)` is identically equal to `verify(foo, times(1))`. The latter would only be useful if you have a test that tests invocation counts other than 1 sometimes.

Reply:
Good point. Updated.